app/models/chargeback_rate.rb

```diff
@@ -3,6 +3,15 @@ class ChargebackRate < ApplicationRecord
   include ReportableMixin
 
   ASSIGNMENT_PARENT_ASSOCIATIONS = [:host, :ems_cluster, :storage, :ext_management_system, :my_enterprise]
+
+  #####################################################################
+  ### Make sure this call back is placed here and before the mixin  ###
+  ### taggings (AssignmentMixin)                                    ###
+  ### The way to ensure that no matter what child records are not   ###
+  ### destoyed prior to parent. Used in the assigned? determination ###
```

Comment:
This message is confusing and doesn't explain the "why" at all.  Maybe this is nicer?

```ruby
# NOTE: ensure_unassigned must occur before the taggings relation is destroyed,
#   since it uses the taggings for its calculation.  The :dependent => :destroy
#   on taggings is part of the destroy callback chain, so we must define this
#   before_destroy here, before that relation is defined, otherwise the callback
#   chain is out of order.  The taggings relation is defined in ar_taggable.rb,
#   and is brought in by the call to acts_as_miq_taggable in the AssignmentMixin
```